elasticsearch.go

```diff
@@ -19,21 +19,38 @@ package elasticsearch
 
 import (
 	"encoding/base64"
+	"encoding/json"
 	"errors"
 	"fmt"
 	"net/http"
 	"net/url"
 	"os"
+	"regexp"
+	"strconv"
 	"strings"
+	"sync"
 	"time"
 
 	"github.com/elastic/go-elasticsearch/v7/esapi"
 	"github.com/elastic/go-elasticsearch/v7/estransport"
 	"github.com/elastic/go-elasticsearch/v7/internal/version"
 )
 
+var (
+	reVersion          *regexp.Regexp
+	useHeaderCheckOnly bool
+	productChecked     bool
+)
+
+func init() {
+	versionPattern := `(\d+).(\d+).(\d+)`
+	reVersion = regexp.MustCompile(versionPattern)
+}
+
 const (
 	defaultURL = "http://localhost:9200"
+	tagline    = "You Know, for Search"
+    unknownProduct = "the client noticed that the server is not Elasticsearch and we do not support this unknown product"
```

Comment:
Are error messages typically lowercased in Go?

Reply:
Yep, they shouldn't end with punctuation either, see https://github.com/golang/go/wiki/CodeReviewComments#error-strings